compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1345,7 +1345,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
     }
 
     def isScalaPhantomSelf(self: tpd.ValDef): Boolean = {
-      if (self.tpt.tpe.derivesFrom(defn.PhantomClass)) {
+      if (self.tpt.tpe.classSymbol eq defn.PhantomClass) {
```

Comment:
I was trying to do a trait that looked like 
```scala
trait Foo { self: Phantom => 
  def x: self.Any = self.asume
}
object Boo extends Phantom with Foo
```
But I ended with some top types with prefix `Foo` instead of `Boo` in places where it should not.

This is mostly intended as a quick fix until I have time to figure out how to re-enable them and be sure they work correctly.